x-pack/plugins/timelines/common/types/timeline/data_provider/index.ts

```diff
@@ -55,7 +55,7 @@ export interface DataProvider {
   /**
    * Additional query clauses that are ANDed with this query to narrow results
    */
-  and: DataProvidersAnd[];
+  and?: DataProvidersAnd[];
```

Comment:
If this was a required field for a data provider and I set it to be an empty array, the auto update PATCH api request would fail and complain about it, I guess we can only nest these 1 level deep?

Reply:
It was surprising that this change only produces one type check failure:

```
 proc [tsc] x-pack/plugins/timelines/public/components/t_grid/helpers.tsx:114:51 - error TS2488: Type 'DataProvidersAnd[] | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.
 proc [tsc]
 proc [tsc] 114       const flatDataProviders = [dataProvider, ...dataProvider.and];
 proc [tsc]                                                       ~~~~~~~~~~~~~~~~
 proc [tsc]
 proc [tsc]
 proc [tsc] Found 1 error.
```

I'm wondering if it's preferable to resolve the unexpected `PATCH` API error instead of changing the type here, because an empty `and` array feels like a valid state.